src/test/java/io/cdap/plugin/aws/sqs/source/SqsSourceConfigTest.java

```diff
@@ -0,0 +1,520 @@
+      Assert.assertEquals(PROPERTY_WAIT_TIME, e.getFailures().get(0).getCauses().get(0)
+        .getAttribute(CauseAttributes.STAGE_CONFIG));
+    }
+  }
+
+  @Test
+  public void testNumberOfMessagesLessThan1() {
+    MockFailureCollector collector = new MockFailureCollector();
+    SqsSourceConfig config = withSqsValidationMock(SqsSourceConfigHelper.newConfigBuilder()
+      .setAuthenticationMethod("Access Credentials")
+      .setAccessId(TEST_ACCESS_ID)
+      .setAccessKey(TEST_ACCESS_KEY)
+      .setSqsEndpoint(TEST_SQS_ENDPOINT)
+      .setRegion(TEST_REGION)
+      .setQueueName(TEST_QUEUE_NAME)
+      .setInterval(1)
+      .setWaitTime(20)
+      .setNumberOfMessages(0)
+      .build(), collector);
+
+    try {
+      config.validate(collector);
+      collector.getOrThrowException();
+    } catch (ValidationException e) {
+      Assert.assertEquals(1, e.getFailures().size());
+      Assert.assertEquals(PROPERTY_NUMBER_OF_MESSAGES, e.getFailures().get(0).getCauses().get(0)
+        .getAttribute(CauseAttributes.STAGE_CONFIG));
+    }
+  }
+
+  @Test
+  public void testNumberOfMessagesGreaterThan10() {
+    MockFailureCollector collector = new MockFailureCollector();
+    SqsSourceConfig config = withSqsValidationMock(SqsSourceConfigHelper.newConfigBuilder()
+      .setAuthenticationMethod("Access Credentials")
+      .setAccessId(TEST_ACCESS_ID)
+      .setAccessKey(TEST_ACCESS_KEY)
+      .setSqsEndpoint(TEST_SQS_ENDPOINT)
+      .setRegion(TEST_REGION)
+      .setQueueName(TEST_QUEUE_NAME)
+      .setInterval(1)
+      .setWaitTime(20)
+      .setNumberOfMessages(11)
+      .build(), collector);
+
+    try {
+      config.validate(collector);
+      collector.getOrThrowException();
+    } catch (ValidationException e) {
+      Assert.assertEquals(1, e.getFailures().size());
+      Assert.assertEquals(PROPERTY_NUMBER_OF_MESSAGES, e.getFailures().get(0).getCauses().get(0)
+        .getAttribute(CauseAttributes.STAGE_CONFIG));
+    }
+  }
+
+  @Test
+  public void testValidateSchemaNull() {
+    MockFailureCollector collector = new MockFailureCollector();
+    SqsSourceConfig config = withSqsValidationMock(SqsSourceConfigHelper.newConfigBuilder()
+      .setAuthenticationMethod("Access Credentials")
+      .setAccessId(TEST_ACCESS_ID)
+      .setAccessKey(TEST_ACCESS_KEY)
+      .setSqsEndpoint(TEST_SQS_ENDPOINT)
+      .setRegion(TEST_REGION)
+      .setQueueName(TEST_QUEUE_NAME)
+      .setInterval(1)
+      .setWaitTime(20)
+      .setNumberOfMessages(10)
+      .setSchema(null)
+      .build(), collector);
+
+    try {
+      config.validate(collector);
+      collector.getOrThrowException();
+    } catch (ValidationException e) {
+      Assert.assertEquals(1, e.getFailures().size());
+      Assert.assertEquals(PROPERTY_SCHEMA, e.getFailures().get(0).getCauses().get(0)
+        .getAttribute(CauseAttributes.STAGE_CONFIG));
+    }
+  }
+
+  @Test
+  public void testValidateFormatInvalidSchema() {
```

Comment:
Maybe add a test for a valid but empty schema?

Reply:
Done

Reply:
The test case for schema with no fields already exists in the code with function name as "testValidateFormatSchemaWithNoFields"